kt/entry-generation/godot-entry-generator/src/main/kotlin/godot/entrygenerator/model/Type.kt

```diff
@@ -0,0 +1,19 @@
+package godot.entrygenerator.model
+
+data class Type(
+    val fqName: String,
+    val kind: TypeKind,
+    val isNullable: Boolean,
+    val supertypes: List<Type>,
+    val arguments: () -> List<Type>
+)
+
+enum class TypeKind {
```

Comment:
Are some of those types just in case of future Godot functionalities?

For example, I hardly see how a type alias can be used for entry gen for now.

Reply:
The idea was so i can check whether a type is a type alias and get the underlying type if necessary. IIRC I'll need this for the intellij plugin to properly resolve if a type is allowed as a default value.
But other than that this could be removed i think.

Reply:
Ho so it's not automatically resolved to the actual underlying type? I thought it was pure syntactic sugar without any technical effect.

Reply:
Nope it's not. The actual underlying type is passed as a super type. So in the case of nesting type aliases, i receive them the same way as a normal class hierarchy.

Reply:
Removed